core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -286,4 +324,40 @@ private[spark] class CoarseMesosSchedulerBackend(
       super.applicationId
     }
 
+  override def doRequestTotalExecutors(requestedTotal: Int): Boolean = {
+    // We don't truly know if we can fulfill the full amount of executors
+    // since at coarse grain it depends on the amount of slaves available.
+    if (executorLimitOption.map(_ != requestedTotal).getOrElse(true)) {
+      logInfo("Capping the total amount of executors to " + requestedTotal)
+    }
+    executorLimitOption = Option(requestedTotal)
```

Comment:
this will always return `Some(int)` I think. You can just use `Some` here so it's clearer at a glance. Same in L360